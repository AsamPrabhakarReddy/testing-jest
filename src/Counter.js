import { useState } from "react";


function Counter()
{
    const [counter,setCount] = useState(0)
    const IncrementFunction = (prev)=>{
        setCount((prev)=> prev + 1)
    }
    const DecrementFunction = (prev)=>{
        setCount((prev)=> prev - 1)
    }
    return (
        <div className="flex items-center gap-10 text-center">
          <div><button data-testid="decre" onClick={DecrementFunction}>-</button></div>
          <div><h1 data-testid="count">{counter}</h1></div>
          <div><button data-testid="incre" onClick={IncrementFunction}>+</button></div>
        </div>
      );
      
}

export default Counter;