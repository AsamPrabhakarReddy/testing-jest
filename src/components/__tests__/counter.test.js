
import { render, fireEvent, screen } from "@testing-library/react";
import renderer from "react-test-renderer"
import Counter from "../../Counter";

describe('Counting functionality Testing == > ', ()=>{
    it('increment testing', ()=>{
        render(<Counter/>)

        const counter = screen.getByTestId('count')
        const increment = screen.getByTestId('incre')

        fireEvent.click(increment)

        expect(counter).toHaveTextContent("1")

    })

    it('decrement counting testing', ()=>{
        render(<Counter/>)

        const counter = screen.getByTestId('count')
        const decrement = screen.getByTestId('decre')

        fireEvent.click(decrement)

        expect(counter).toHaveTextContent("-1")
    })
})


describe('using snapshot ', ()=>{
    it("snapshot testing",()=>{
        const tree = renderer.create(<Counter/>).toJSON();
        expect(tree).toMatchSnapshot();

    })
})