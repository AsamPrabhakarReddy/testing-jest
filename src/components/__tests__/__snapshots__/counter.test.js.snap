// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`using snapshot  snapshot testing 1`] = `
<div
  className="flex items-center gap-10 text-center"
>
  <div>
    <button
      data-testid="decre"
      onClick={[Function]}
    >
      -
    </button>
  </div>
  <div>
    <h1
      data-testid="count"
    >
      0
    </h1>
  </div>
  <div>
    <button
      data-testid="incre"
      onClick={[Function]}
    >
      +
    </button>
  </div>
</div>
`;
